Product Management Service/Code/product-service/src/main/java/com/shopkeeper/productservice/service/ProductService.java

```diff
@@ -5,32 +5,46 @@
 import java.util.List;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
-import com.shopkeeper.productservice.dao.ProductServiceRepository;
+import com.shopkeeper.productservice.dao.ProductRepository;
 import com.shopkeeper.productservice.dto.Product;
 
 @Service
 public class ProductService {
 
-  @Autowired ProductServiceRepository productRepository;
+  @Autowired ProductRepository productRepository;
 
-  public List<Product> getAllProducts() {
+  public ResponseEntity<List<Product>> isCategoryPresent(String category) {
+    if (category != null && !category.isEmpty()) {
+      return getProductByCategory(category);
+    } else {
+      return getAllProducts();
+    }
+  }
+
+  public ResponseEntity<List<Product>> getAllProducts() {
     List<Product> products = new ArrayList<Product>();
     productRepository.findAll().forEach(products::add);
-    return products;
+    return new ResponseEntity<>(products, HttpStatus.OK);
   }
 
-  public Product getProductById(Integer id) {
-    return productRepository.findById(id).orElse(null);
+  public ResponseEntity<Product> getProductById(Integer id) {
+    Product product = productRepository.findById(id).orElse(null);
+    if (product != null) {
+      return new ResponseEntity<>(product, HttpStatus.OK);
+    } else {
+      return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
```

Comment:
these can either be one public method, or 1 public and 2 private methods, but not 3 public methods for sure.